### PR TITLE
[IA-4805] bug fix in zombie monitor/resource validator caused from last PR

### PR DESCRIPTION
**Pull request by Qi77Qi**
Tested resource validator and zombie monitor locally

### CODE REVIEW

File: core/src/main/scala/com/broadinstitute/dsp/DbReaderImplicits.scala [hunk 1]
```diff
@@ -113,9 +113,18 @@ object DbReaderImplicits {
...
-            )
+            AzureCloudContext.fromString(cloudContextDb) match {
+              case Left(value) =>
+                throw new RuntimeException(
+                  s"${value} is not valid azure cloud context"
+                )
+              case Right(value) =>
+                KubernetesCluster(id, name, CloudContext.Azure(value), location)
+            }
+//           TODO: throw exception once we fix query in zombie monitor and resource validator
```
Comment (mlilynolting) [inline core/src/main/scala/com/broadinstitute/dsp/DbReaderImplicits.scala:124]: Resource Validator does not support Azure yet, but Zombie Monitor does currently handle AKS clusters (see https://github.com/broadinstitute/leonardo-cron-jobs/blob/2fb1c0f194555c36269ed0888ff26762c1abcd14/zombie-monitor/src/main/scala/com/broadinstitute/dsp/zombieMonitor/DeletedKubernetesClusterChecker.scala#L40). So I think that the exception is probably inappropriate here.

**Review by mlilynolting — COMMENTED** [review]
This looks like a good fix. I'm unsure why the diff shows changes in DbReader; weren't those already done in your original PR?

**Comment by mlilynolting** [comment]
Jira ticket is https://broadworkbench.atlassian.net/browse/IA-4805

**Review by mlilynolting — APPROVED** [review]

**Comment by Qi77Qi** [comment]
@mlilynolting thanks for taking this to the finish line!

**Pull request merged**